test/groovy/util/JenkinsShellCallRule.groovy

```diff
@@ -73,6 +79,13 @@ class JenkinsShellCallRule implements TestRule {
                             }
                             if(result instanceof Closure) result = result()
                             return result
+                        } else if (m.returnStatus) {
```

Comment:
Should we try to improve the control flow? Above we do $something in case `m.returnStatus` is set (or (not xor) `m.returnStdOut` is set). This might end-up in a `return result`. So the code here doesn't get executed.

Is there really the need for introducing a second entity for handling return status? Wouln't it be possible with the current coding also to handle return status by using the `returnValue` also here. In case 'yes': maybe we should generalize that name.

Side note only: What about splitting off changes to test-tooling from introducing new steps in order to have clear focuses?

Reply:
You are right, we need to improve here.
One thing which is not handled so far is that we get a status 0 unless set differently for returnStatus.